tfhe/src/shortint/engine/client_side.rs

```diff
@@ -300,7 +300,13 @@ impl ShortintEngine {
 
     pub fn decrypt(&mut self, client_key: &ClientKey, ct: &Ciphertext) -> EngineResult<u64> {
         self.decrypt_message_and_carry(client_key, ct)
-            .map(|message_and_carry| message_and_carry % ct.message_modulus.0 as u64)
+            .map(|message_and_carry| {
+                let padding_bit =
+                    message_and_carry / (ct.carry_modulus.0 * ct.message_modulus.0) as u64;
+
+                assert_eq!(padding_bit, 0);
```

Comment:
so that's an interesting idea but I would probably see something like an Ok(val) if padding bit is 0 and an Err(val) with a nice error formatting for the Err variant

I don't know if asserting here is a good idea, decrypting is not failing per se